fix for broken template mapping json

Munawwar opened the pull request:
Seems like the mapping part of json file needs to be wrapped into a "type" (which can be any name). This is on elasticsearch 6. I am not sure if older elastic used to accept without it.

e.g.
```
"mapping": {
  "asdasd": {
     "properties": { .... }
  }
}
```

vanthome commented on index-template-mapping.json:
```diff
@@ -8,15 +8,16 @@
-      "fields": {
-        "dynamic": true,
-        "properties": { }
+    "winston-elasticsearch-test-log-type": {
```
I guess this is the name of the type that will then appear in Kibana? 
What about `_source` why have you removed that?

Munawwar replied:
Ah I think I realize now why the template mapping didn't work. I use ES 6. And you probably use ES 7... ES 7 removed mapping type and the template json structure changed for ES 7. So the existing template json works on ES 7 and my template works on ES 6.
I will re-implement my fix to have two templates and auto-detect the ES version before the tests run and use the right one (it's just for the tests to work I believe.. right?).

And about the template name appearing in kibana.. yes it will appear as the _type field of each document. ES documentation uses "_doc" as type name when you don't want to give a fancy type name. I'll do that as well.

Munawwar replied:
And about _source..
1. source is enabled by default anyway. So it's redundant to mention it. But we can add it if you want to be explicit about it.
2. the syntax should've been "_source": { "enabled": true } and not "_source": true
ES 6 throws an error with that. (I think it's not valid in ES 7 either from the docs -> https://www.elastic.co/guide/en/elasticsearch/reference/7.8/mapping-source-field.html)

vanthome replied:
ok, agreed

Munawwar commented:
Ok. I've changed a couple of things
1. There had to be two template.json files - one for elasticsearch version 7 & above, and another one for elasticsearch version 6 & below

2. Which means there has to be a way to know which ES version is being used. I introduced elasticsearchVersion field. As I think making a request for a not-so-common case (i.e. ensureMappingTemplate = true + mappingTemplate = undefined combination) is a bit of an overkill.. what do you think?

3. However for tests, to test without specifying the template json file like it is now, I had to make a call to ES to find out the version.

4. There is a bug in how log writing starts before ensuring template mapping succeeds. So fixed that as well.
 
5. Another blocking bug I found on my setup is that the health check code fails with error 408 when you have more than one node in your elasticsearch cluster. I guess because the health check is hard-coded to '1'. It should rather be '>=1'.

vanthome commented:
can you please rebase it?

Munawwar commented:
@vanthome Done.

teyou commented:
Pls help update `index.d.ts` as well as there are more options in `ElasticsearchTransportOptions`.

related : https://github.com/vanthome/winston-elasticsearch/issues/134

Munawwar commented:
Added it
